### PR TITLE
V5.0 Rebranding UtilityConnect -> Connect

### PR DESCRIPTION
**Pull request by xelzi**
Closes: https://arcadiapower.atlassian.net/browse/EN-796
BLOCKED UNTIL NEW CONNECT BUNDLE IS HOSTED ON CDN** 

1. Rebranding in README
2. Update Changelog
3. Update Package JSON with new package name + version -- should be able to republish to npm without impacting current `utility-connect-react` package
4. Update Hook implementation to use new naming scheme

### CODE REVIEW

File: src/use-connect.js [hunk 1]
```diff
@@ -4,21 +4,21 @@ import useScript from 'react-script-hook';
 const MAX_POLL_NUMBER = 3;
 const POLLING_INTERVAL = 100;
 
-const src = 'https://utility-connect-main.prod.arcadia.com/dist/v3.js';
+const src = 'https://connect-main.prod.arcadia.com/dist/v3.js';
```
Comment (xelzi) [inline src/use-connect.js:7]: Need to make double check this URL and version after `Connect` bundle changes are made 👍

File: src/use-connect.js [hunk 2]
```diff
@@ -35,9 +35,9 @@ export const useUtilityConnect = () => {
   });
 
   const pollForFactory = useCallback(() => {
-    const { _ArcadiaUtilityConnect } = window;
-    if (_ArcadiaUtilityConnect) {
-      setFactory(_ArcadiaUtilityConnect);
+    const { _ArcConnect } = window;
+    if (_ArcConnect) {
+      setFactory(_ArcConnect);
```
Comment (xelzi) [inline src/use-connect.js:40]: need to change this here: https://github.com/ArcadiaPower/utility-connect/blob/main/src/index.js

File: README.md [hunk 3]
```diff
@@ -1,43 +1,40 @@
...
-
-If the Utility Connect Token is instantiated with an exisiting `utility_credential_id`, then Utility Connect will update the existing UtilityCredential. This is referred to in the documentation as 'update' mode. Otherwise, Utility Connect will create a new, unique UtilityCredential which is referred to as 'create' mode.
-
-Utility Connect automatically infers the correct user flow (either 'update' or 'create') from the Utility Connect Token. When in 'create' mode, the user will pass through the entire Utility Connect flow, including the Consent Pane (accepting terms of service), Utility Pane (selecting a utility), and Credential Pane (entering utility credentials). When in 'update' mode, the user will be immediately placed on the Credentials Pane in order to reduce user friction. 
+A Connect Token is required to instantiate Connect. See [Creating Connect Tokens](https://developers.arcadia.com/#section/Authentication/Connect) for instructions. Note that a general Access Token will not work for this purpose.
```
Comment (xelzi) [inline README.md:23]: link will be broken til we release the doc changes

File: README.md [hunk 4]
```diff
@@ -1,43 +1,40 @@
-# @arcadia-eng/utility-connect-react
+# @arcadia-eng/connect-react
 
-Arcadia's general Developer Platform API documentation can be found at [developers.arcadia.com](https://developers.arcadia.com). The purpose of this package is to embed a user-facing interface where your customers can securely enter their utility credentials. The submission of their utility-related credentials will create UtilityCredentials and UtilityAccounts that can be managed through the Platform API.
+Arcadia's general Developer Platform API documentation can be found at [developers.arcadia.com](https://developers.arcadia.com). The purpose of this package is to embed a user-facing interface where your customers can securely enter their utility credentials. The submission of their utility-related credentials will create UtilityCredentials and UtilityAccounts that can be managed through Arc.
```
Comment (KerriDevine) [inline README.md:3]: > Arcadia's general Developer Platform API

This should probably be `Arc API documentation`.

File: README.md [hunk 5]
```diff
@@ -105,35 +102,35 @@ Data is expected in the following format:
...
-- `'pending'`: the user submitted credentials but closed Utility Connect before the component could get updated with the result
+- `'timed_out'`: Connect times out awaiting verification of the utility credentials. In this case, if clients later receive the `UtilityCredentialRejected` webhook or confirm rejection through the UtilityCredentials API endpoint, client will likely want to redirect the user back Connect (via in-app notifications or emails). If the user needs to update their credentials, the client should fetch a `UtilityConnectToken` with the pre-existing `utility_credential_id` so that Connect opens in "update" mode.
+- `'pending'`: the user submitted credentials but closed Connect before the component could get updated with the result
 - `'no_submit'`: the user never submitted their credentials
-- `'error'`: there was an API error during the Utility Connect flow and the user clicked a button to confirm as such and close Utility Connect
+- `'error'`: there was an API error during the Connect flow and the user clicked a button to confirm as such and close Connect
```
Comment (KerriDevine) [inline README.md:133]: this error description was a little confusing even before this update. Maybe we could try, "there was an API error during the Connect flow. User clicked to confirm the error and close Connect."

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "@arcadia-eng/utility-connect-react",
-  "version": "4.0.3",
-  "description": "React tool for integrating with Arcadia's utility connect service",
+  "name": "@arcadia-eng/connect-react",
+  "version": "5.0.0",
+  "description": "React tool for integrating with Arc Connect",
```
Comment (KerriDevine) [inline package.json:4]: Technically, `Arc Connect` is the name of the React tool, no? Should this be "React component for integrating with Arc Plug"? 😕 @johnnyman727

**Review by KerriDevine — COMMENTED** [review]
Left a few minor comments on the README. Thank you for digging in!

File: README.md (hunk 5 above)
Comment (clifff) [reply README.md:133]:
```suggestion
- `'error'`: there was an API error during the Connect flow. User clicked to confirm the error and closed Connect.
```

File: src/use-connect.js (hunk 1 above)
Comment (clifff) [reply src/use-connect.js:7]: Should be all set!

File: package.json (hunk 6 above)
Comment (clifff) [reply package.json:4]: "React package for implementing Arc Connect" maybe? In my mind the name of the widget is `Connect` but it's part of the Arc ecosystem so it is situationally appropriate to preface it with that

**Review by clifff — APPROVED** [review]
Looks solid! Is there a good way to test this locally before merging? Updates only go out if we specifically push to NPM so should be safe...

File: src/use-connect.js (hunk 2 above)
Comment (xelzi) [reply src/use-connect.js:40]: DONE

**Comment by xelzi** [comment]
Alright -- tested this bundle in the demo app and everything looks good ⚡

**Pull request merged**